e2e-tests/ns-mode/run

```diff
@@ -0,0 +1,46 @@
+#!/bin/bash
+
+set -o errexit
+set -o xtrace
+
+test_dir=$(realpath $(dirname $0))
+. ${test_dir}/../functions
+
+function main {
+	create_namespace $namespace
+	# run operator in the same namespace as cr (default configuration)
+	deploy_operator
+
+	desc 'create first PG cluster'
+	cluster="some-name"
+	spinup_pgcluster "${cluster}" "${src_dir}/deploy/cr.yaml"
+
+	destroy_operator
+
+	desc 'dynymic operator mode. Should create namespace for cluster by itself'
+	ns="${test_name}-${RANDOM}"
+	create_namespace ${ns}
+	export OPERATOR_NS="${ns}"
+
+	deploy_operator '' "${src_dir}/deploy/operator.yaml" 'dynamic'
+	spinup_pgcluster "${cluster}" "${src_dir}/deploy/cr.yaml"
+	destroy_operator
+	destroy_namespace $OPERATOR_NS
+	unset OPERATOR_NS
+
+	desc 'readonly operator mode. Should only watch namespace for cluster. No namespace creation or removal'
+	create_namespace $namespace
+
+	deploy_operator '' "${src_dir}/deploy/operator.yaml" 'disabled'
+	spinup_pgcluster "${cluster}" "${src_dir}/deploy/cr.yaml"
+
+	create_namespace "${namespace}-disabled" 'true'
+	export OPERATOR_NS="${namespace}-disabled"
+	deploy_operator '' "${src_dir}/deploy/operator.yaml" 'disabled' '' "${namespace}-disabled"
+	spinup_pgcluster "${cluster}" "${src_dir}/deploy/cr.yaml" '' '' '' '' '' '' '' '' "${namespace}-disabled"
```

Comment:
I believe we need to change the way of how we pass the arguments. Key-value approach via the configmap could be a way out here

Reply:
ok np, we can do it via separate PR